lib/actions-oclif/config/generate.ts

```diff
@@ -0,0 +1,239 @@
+			description: 'generate a fresh device key for the device',
+		}),
+		output: flags.string({
+			description: 'path of output file',
+			char: 'o',
+		}),
+		// Options for non-interactive configuration
+		network: flags.string({
+			description: 'the network type to use: ethernet or wifi',
+			options: ['ethernet', 'wifi'],
+		}),
+		wifiSsid: flags.string({
+			description:
+				'the wifi ssid to use (used only if --network is set to wifi)',
+		}),
+		wifiKey: flags.string({
+			description:
+				'the wifi key to use (used only if --network is set to wifi)',
+		}),
+		appUpdatePollInterval: flags.string({
+			description:
+				'how frequently (in minutes) to poll for application updates',
+		}),
+		help: cf.help,
+	};
+
+	public static authenticated = true;
+
+	public async run() {
+		const { flags: options } = this.parse<FlagsDef, {}>(ConfigGenerateCmd);
+
+		const balena = getBalenaSdk();
+		const { tryAsInteger } = await import('../../utils/validation');
+
+		await this.validateOptions(options);
+
+		// Get device | application
+		let resource;
+		if (options.device != null) {
+			resource = await balena.models.device.get(tryAsInteger(options.device));
+		} else {
+			resource = await balena.models.application.get(options.application!);
+		}
+
+		const deviceType = options.deviceType || resource.device_type;
+
+		const deviceManifest = await balena.models.device.getManifestBySlug(
+			deviceType,
+		);
+
+		// Check compatibility if application and deviceType provided
+		if (options.application && options.deviceType) {
+			const appDeviceManifest = await balena.models.device.getManifestBySlug(
+				resource.device_type,
+			);
+
+			const helpers = await import('../../utils/helpers');
+			if (
+				!helpers.areDeviceTypesCompatible(appDeviceManifest, deviceManifest)
+			) {
+				throw new balena.errors.BalenaInvalidDeviceType(
+					`Device type ${options.deviceType} is incompatible with application ${options.application}`,
+				);
+			}
+		}
+
+		// Prompt for values
+		// Pass params as an override: if there is any param with exactly the same name as a
+		// required option, that value is used (and the corresponding question is not asked)
+		const answers = await getCliForm().run(deviceManifest.options, {
+			override: options,
+		});
+		answers.version = options.version;
+
+		// Generate config
+		const { generateDeviceConfig, generateApplicationConfig } = await import(
+			'../../utils/config'
+		);
+
+		let config;
+		if ((resource as Device).uuid != null) {
+			config = await generateDeviceConfig(
+				resource as Device & { belongs_to__application: PineDeferred },
+				options.deviceApiKey || options['generate-device-api-key'] || undefined,
+				answers,
+			);
+		} else {
+			answers.deviceType = deviceType;
+			config = await generateApplicationConfig(
+				resource as Application,
+				answers,
+			);
+		}
+
+		// Output
+		if (options.output != null) {
+			await require('fs').promises.writeFile(
```

Comment:
Can you make this
```suggestion
			const fs = await import('fs');
			await fs.promises.writeFile(
```
so that we benefit from type-checking (`require` returns `any` for `.ts` files)

Reply:
Ah missed that.